<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Dr.&nbsp;Mario Reutter (slides adapted from Dr.&nbsp;Lea Hildebrandt)">

<title>Biostatistics - 07 Hypothesis Testing</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating slimcontent">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./W7_Hypothesis.html">07 Hypothesis Testing</a></li></ol></nav>
        <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Biostatistics</a> 
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W1_Intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">01a Introduction to Biostatistics</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W1b_IntroR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">01b Intro to R</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W2_Models.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">02 Models</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W3_DataWranglingR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">03 Data Wrangling</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W4_DataVizR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">04 Data Visualization</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W5_Sampling.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">05 Probabilities &amp; Sampling</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W6_ProbabilityR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">06 Probability &amp; Sampling in R</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W7_Hypothesis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">07 Hypothesis Testing</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W8_ModelingRelationships.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">08 Modeling Relationships</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W9_FirstAnalysisR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">09 Statistical Analyses in R</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W10_GLM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">10 The General Linear Model &amp; Multivariate Statistics</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W11_GLMinR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">11 ANOVA &amp; the General Linear Model in R</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W12_LMM.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">12 Linear Mixed Models</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W13_RMarkdown.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">13 R Markdown</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./W15_RepResearchR.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">15 Reproducible Research</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        
    <div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="W7_Hypothesis.html"><i class="bi bi-file-slides"></i>RevealJS</a></li></ul></div></div>
<!-- main -->
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">07 Hypothesis Testing</h1>
<p class="subtitle lead"></p><div class="line-block">Julius-Maximilians-University Würzburg<br>
Course: “Biostatistics”<br>
Translational Neuroscience</div><p></p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Dr.&nbsp;Mario Reutter (slides adapted from Dr.&nbsp;Lea Hildebrandt) </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="hypothesis-testing" class="level1 page-columns page-full">
<h1>Hypothesis Testing</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode css code-with-copy"><code class="sourceCode css"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>code<span class="fu">.sourceCode</span> {</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>  <span class="kw">font-size</span><span class="ch">:</span> <span class="dv">1.4</span><span class="dt">em</span><span class="op">;</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>div<span class="fu">.cell-output-stdout</span> {</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="kw">font-size</span><span class="ch">:</span> <span class="dv">1.4</span><span class="dt">em</span><span class="op">;</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<style type="text/css">
code.sourceCode {
  font-size: 1.4em;
}

div.cell-output-stdout {
  font-size: 1.4em;
}
</style>
</div>
<p>Remember, the three goals of statistics: Describe, Decide, Predict.</p>
<p>We will now turn to deciding whether a particular hypothesis is supported by the data.</p>
<p>. . .</p>
<p>We will cover <strong>null hypothesis significance testing (NHST)</strong>, which is the main technique you will encounter in publications.</p>
<p>. . .</p>
<p>However, NHST is also criticized (see chapter 18 of ST21)! It is often used mindlessly and often misunderstood.</p>
<div class="notes">
<p>learning to use and interpret results is crucial to understand research</p>
</div>
<section id="null-hypothesis-significance-testing" class="level2">
<h2 class="anchored" data-anchor-id="null-hypothesis-significance-testing">Null Hypothesis Significance Testing</h2>
<p>Example: We have two groups (treatment and control). We also have a hypothesis: The treatment group has lower scores on measure X (e.g., symptoms). We have the data (X for both groups), now what?</p>
<div class="incremental">
<ul class="incremental">
<li><p>We take the hypothesis (treatment = lower X than control) and negate it (Treatment not lower/equal X compared to control). This is our <strong>null hypothesis</strong>.</p></li>
<li><p>Then we look at the data and determine how likely they would be if the null hypothesis were true.<br>
I.e., we want to know the conditional probability: <span class="math inline">\(P(Data|H_0)\)</span></p></li>
<li><p>If the data are <em>very unlikely</em> we reject the null hypothesis in favor of the <strong>alternative hypothesis</strong> (our hypothesis).</p></li>
<li><p>(If the data are not very unlikely, we stick with - or fail to reject - the null hypothesis.)</p></li>
</ul>
</div>
<div class="notes">
<p>How would you compare the two groups? Calculate the likelihood that there is a reduction in X between the groups? No, more complicated! And counterintuitive!</p>
</div>
</section>
<section id="the-process-of-nhst" class="level2">
<h2 class="anchored" data-anchor-id="the-process-of-nhst">The Process of NHST</h2>
<p>To be more precise, we can break down the process of null hypothesis testing in six steps:</p>
<div class="incremental">
<ol class="incremental" type="1">
<li><p>Formulate a hypothesis that embodies our prediction (<em>before seeing the data</em>)</p></li>
<li><p>Specify null and alternative hypotheses that reflect the hypothesis formulated in step 1</p></li>
<li><p>Collect some data relevant to the hypothesis</p></li>
<li><p>Fit a model to the data that represents the alternative hypothesis and compute a test statistic</p></li>
<li><p>Compute the probability of the observed value of that statistic assuming that the null hypothesis is true</p></li>
<li><p>Assess the “statistical significance” of the result</p></li>
</ol>
</div>
<p>. . .</p>
<p>Let’s go through these steps, using the NHANES dataset and the research question: <em>Is physical activity related to body mass index (BMI)</em>?</p>
</section>
<section id="step-1-formulate-a-hypothesis-of-interest" class="level2">
<h2 class="anchored" data-anchor-id="step-1-formulate-a-hypothesis-of-interest">Step 1: Formulate a Hypothesis of Interest</h2>
<p>Hypothesis:</p>
<p>. . .</p>
<p>“BMI is greater for people who do not engage in physical activity than for those who do.”</p>
<div class="notes">
<p>ask for hypothesis?</p>
</div>
</section>
<section id="step-2-specify-the-null-and-alternative-hypotheses" class="level2">
<h2 class="anchored" data-anchor-id="step-2-specify-the-null-and-alternative-hypotheses">Step 2: Specify the Null and Alternative Hypotheses</h2>
<p>The null hypothesis (<span class="math inline">\(H_0\)</span>) is the baseline against which we test our hypothesis of interest.</p>
<p>The alternative hypothesis (<span class="math inline">\(H_A\)</span>) describes what we expect if there is an effect.</p>
<p>NHST works under the assumption that the <span class="math inline">\(H_0\)</span> is true (unless the evidence shows otherwise).</p>
<p>. . .</p>
<p>We also have to decide whether we want to test a non-directional (<span class="math inline">\(A \neq B\)</span>) or directional (<span class="math inline">\(A&gt;B\)</span> <strong>or</strong> <span class="math inline">\(A&lt;B\)</span>) hypothesis.</p>
<p>What do we specify if we hypothesize that “BMI is greater…”?</p>
<p>. . .</p>
<p><span class="math inline">\(H_0 = BMI_{active} \ge BMI_{inactive}\)</span></p>
<p><span class="math inline">\(H_A = BMI_{active} &lt; BMI_{inactive}\)</span></p>
<div class="notes">
<p>Test against <span class="math inline">\(H_0\)</span>: What would we expect the data to look like if there was no effect?</p>
<p>Non-directional: no direction! :D Directional: prior knowledge</p>
</div>
</section>
<section id="step-3-collect-data" class="level2">
<h2 class="anchored" data-anchor-id="step-3-collect-data">Step 3: Collect Data</h2>
<p>For this example, we sample 250 individuals from the NHANES dataset.</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'cowplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:patchwork':

    align_plots</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:lubridate':

    stamp</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attache Paket: 'MASS'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:patchwork':

    area</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Das folgende Objekt ist maskiert 'package:dplyr':

    select</code></pre>
</div>
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>Summary of BMI data for active versus inactive individuals</caption>
<thead>
<tr class="header">
<th style="text-align: left;">PhysActive</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">sd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">No</td>
<td style="text-align: right;">131</td>
<td style="text-align: right;">30.1942</td>
<td style="text-align: right;">8.9851</td>
</tr>
<tr class="even">
<td style="text-align: left;">Yes</td>
<td style="text-align: right;">119</td>
<td style="text-align: right;">26.6386</td>
<td style="text-align: right;">5.2499</td>
</tr>
</tbody>
</table>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="W7_Hypothesis_files/figure-html/unnamed-chunk-2-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="step-4-fit-a-model" class="level2">
<h2 class="anchored" data-anchor-id="step-4-fit-a-model">Step 4: Fit a Model</h2>
<p>We want to compute a <strong>test statistic</strong> that helps us decide whether to reject <span class="math inline">\(H_0\)</span> or not.</p>
<p>. . .</p>
<p>The model we fit needs to quantify (= provide the test statistic) the amount of evidence in favor of <span class="math inline">\(H_A\)</span> relative to the variability of the data.</p>
<p>The test statistic will have a probability distribution, allowing us to determine how likely our observed value of the statistic is under <span class="math inline">\(H_0\)</span>.</p>
<p>. . .</p>
<p><strong>In general, we want to relate an effect (e.g., a mean or a difference of means) to the amount of uncertainty in the data (e.g., the SEM).</strong></p>
<p>. . .</p>
<p>In the example, we need a test statistic that tests the difference between two means (we have one BMI mean for each group): The <em>t</em> statistic.</p>
<p><span class="math display">\[t = \frac{\bar{X_1} - \bar{X_2}}{\sqrt{\frac{S_1^2}{n_1} + \frac{S_2^2}{n_2}}}\]</span></p>
<p><span class="math inline">\(\bar{X_1}\)</span> and <span class="math inline">\(\bar{X_2}\)</span> are the means of the two group, <span class="math inline">\(S_1^2\)</span> and <span class="math inline">\(S_2^2\)</span> are the estimated variances of the groups, <span class="math inline">\(n_1\)</span> and <span class="math inline">\(n_2\)</span> are the sizes of the two groups.</p>
<p>. . .</p>
<p><span class="math inline">\(\sqrt{\frac{S_1^2}{n_1} + \frac{S_2^2}{n_2}}\)</span> is something like the <em>pooled</em> (“averaged”) SEM of both groups.</p>
<div class="notes">
<p>The <em>t</em> statistic is appropriate for comparing the means of two groups when the sample sizes are relatively small and the population standard deviation is unknown.</p>
<p><span class="math inline">\(\sqrt{\frac{S_1^2}{n_1}}\)</span> alone would be the SEM of subsample 1. But we have to add first before we take the square root.</p>
<p>Reason for adding: The variance of a difference between (or sum of) two independent variables is the sum of the variances of each individual variable (<span class="math inline">\(Var(A−B)=Var(A)+Var(B)\)</span>)</p>
<p>one can view the the <em>t</em> statistic as a way of quantifying how large the difference between groups is in relation to the sampling variability of the difference between means</p>
</div>
</section>
<section id="the-t-distribution" class="level2">
<h2 class="anchored" data-anchor-id="the-t-distribution">The <em>t</em> Distribution</h2>
<p>The <em>t</em> statistic is distributed according to the <em>t</em> distribution, which looks similar to a normal distribution (the more degrees of freedom, the more “normal”).</p>
<p><strong>Degrees of freedom</strong> for the <em>t</em> test: <span class="math inline">\(observations - 2\)</span> = <span class="math inline">\(n_1 + n_2 - 2\)</span> (when the groups are the same size).</p>
<p>Degrees of freedom: values that can freely vary when estimating parameters. Usually sample size minus values that you already calculated (e.g.&nbsp;means for the test statistic).</p>
<p>. . .</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">tibble</span>(<span class="at">z =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">4</span>, .<span class="dv">1</span>), </span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>       <span class="at">t1 =</span> <span class="fu">dt</span>(z, <span class="dv">1</span>),</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>       <span class="at">t10 =</span> <span class="fu">dt</span>(z, <span class="dv">10</span>),</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>       <span class="at">t100 =</span> <span class="fu">dt</span>(z, <span class="dv">100</span>),</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">normal =</span> <span class="fu">dnorm</span>(z)) <span class="sc">%&gt;%</span> </span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span>z, <span class="at">names_to =</span> <span class="st">"distribution"</span>, <span class="at">values_to =</span> <span class="st">"density"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>z, <span class="at">y=</span>density, <span class="at">color=</span>distribution)) <span class="sc">+</span> <span class="fu">geom_line</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_color_viridis_d</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="W7_Hypothesis_files/figure-html/unnamed-chunk-3-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>If the group sizes are unequal: <span class="math inline">\(\mathrm{d.f.} = \frac{\left(\frac{S_1^2}{n_1} + \frac{S_2^2}{n_2}\right)^2}{\frac{\left(S_1^2/n_1\right)^2}{n_1-1} + \frac{\left(S_2^2/n_2\right)^2}{n_2-1}}\)</span></p>
<p>Fortunately, R does all these calculations for us :)</p>
<div class="notes">
<p>DF: we have calculated two means and have thus given up two DFs (these are fixed already)</p>
<p>Will be smaller if sample sizes unequal (here 241.12 vs 248 for equal)</p>
<p>Degrees of freedom are the number of independent values that a statistical analysis can <a href="https://statisticsbyjim.com/glossary/estimator/">estimate</a>. You can also think of it as the number of values that are free to vary as you estimate <a href="https://statisticsbyjim.com/glossary/parameter/">parameters</a></p>
</div>
</section>
<section id="step-5-determine-the-probability-of-the-observed-result-under-the-null-hypothesis" class="level2">
<h2 class="anchored" data-anchor-id="step-5-determine-the-probability-of-the-observed-result-under-the-null-hypothesis">Step 5: Determine the Probability of the Observed Result under the Null Hypothesis</h2>
<p>We do not check likelihood of the alternative distribution or likelihood that the null hypothesis is true, but rather:</p>
<p><em>How likely is it, given that we assume</em> <span class="math inline">\(H_0\)</span> <em>is true, to observe a statistic at least as extreme as the one we observed.</em></p>
<p>--&gt; We need to know the distribution of the expected statistic, assuming <span class="math inline">\(H_0\)</span> is true. Then we can calculate how (un-)likely it is to find the statistic (or a more extreme value) we found in our data.</p>
<p>. . .</p>
<div class="cell">
<div class="cell-output cell-output-stderr">
<pre><code>Warning: A numeric `legend.position` argument in `theme()` was deprecated in ggplot2
3.5.0.
ℹ Please use the `legend.position.inside` argument of `theme()` instead.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="W7_Hypothesis_files/figure-html/unnamed-chunk-4-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="notes">
<p>counter-intuitive: We check the nulldistribution not the one of <span class="math inline">\(H_A\)</span>! But we also don’t check how likely it is that <span class="math inline">\(H_0\)</span> is true, but rather the likelihood under the null hypothesis of observing a statistic at least as extreme as the one we observed.</p>
<p>at least as extreme: Prob of each particular value = 0</p>
<p>try to find out how weird statistic found is (or weirder) –&gt; count all weird(er) possibilities</p>
</div>
</section>
<section id="a-simple-example" class="level2">
<h2 class="anchored" data-anchor-id="a-simple-example">A Simple Example</h2>
<p>Is a coin biased if we flip a coin 100x and we get 70 heads?</p>
<p><span class="math inline">\(H_0: P(heads) \le 0.5\)</span> and <span class="math inline">\(H_A: P(heads) &gt; 0.5\)</span></p>
<p>Test statistic = number of heads counted.</p>
<p>How likely is it that we would observe 70 or more heads if the coin is unbiased (chance of 50% for heads)?</p>
<p>. . .</p>
<p>If we flip a (fair) coin 100 times, we would get the following distribution (100000 replications):</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="W7_Hypothesis_files/figure-html/unnamed-chunk-5-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>It is very unlikely to get 70 heads if the coin is fair!</p>
<div class="notes">
<p>fair coin: null distribution</p>
</div>
</section>
<section id="p-value" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="p-value">P-Value</h2>
<p>Let’s go back to out BMI example.</p>
<p>We first need to calculate the <em>t</em> statistic:</p>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>Summary of BMI data for active versus inactive individuals</caption>
<thead>
<tr class="header">
<th style="text-align: left;">PhysActive</th>
<th style="text-align: right;">N</th>
<th style="text-align: right;">mean</th>
<th style="text-align: right;">sd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">No</td>
<td style="text-align: right;">131</td>
<td style="text-align: right;">30.1942</td>
<td style="text-align: right;">8.9851</td>
</tr>
<tr class="even">
<td style="text-align: left;">Yes</td>
<td style="text-align: right;">119</td>
<td style="text-align: right;">26.6386</td>
<td style="text-align: right;">5.2499</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math display">\[t = \frac{\bar{X_1} - \bar{X_2}}{\sqrt{\frac{S_1^2}{n_1} + \frac{S_2^2}{n_2}}}\]</span></p>
<p><span class="math display">\[t = \frac{30 - 27}{\sqrt{\frac{9^2}{131} + \frac{5.2^2}{119}}}\]</span></p>
<p><span class="math display">\[t = 3.86\]</span></p>
<p>. . .</p>
<p>The question is: What is the likelihood that we would find a <em>t</em> statistic of this size or more extreme given the number of degrees of freedom and if the true difference between the groups is zero.</p>

<div class="no-row-height column-margin column-container"><div class="margin-aside">
<p><span class="math inline">\(t\)</span> statistics if we use the unrounded values!</p>
</div></div></section>
<section id="p-value-2" class="level2">
<h2 class="anchored" data-anchor-id="p-value-2">P-Value 2</h2>
<p>We can use the <em>t</em> distribution to calculate this probability. We just need the degrees of freedom, which are <span class="math inline">\(DF = 241.12\)</span>, and we can then use all these values (e.g.&nbsp;in a function in R):</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co">#cumulative probability distribution of the student's t distribution</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pt</span>(<span class="fl">3.86</span>, <span class="at">df=</span><span class="fl">241.12</span>, <span class="at">lower.tail =</span> F)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 7.282672e-05</code></pre>
</div>
</div>
<p>This small probability tells us that our observed t value is relatively unlikely if <span class="math inline">\(H_0\)</span> is really true.</p>
<p>. . .</p>
<p>This is the <em>p</em>-Value for a directional hypothesis. In this case, we only looked at the upper tail probability. With a non-directional hypothesis, we would want to account for both tail probabilities, i.e.&nbsp;how likely it is that a <span class="math inline">\(t &gt; 3.86\)</span> OR <span class="math inline">\(t &lt; -3.86\)</span> is found. In this case, we can simply multiply the <em>p</em>-Value found above by 2 (since it is a symmetric distribution):</p>
<p><span class="math inline">\(p = 0.000145\)</span></p>
</section>
<section id="p-value-using-randomization" class="level2">
<h2 class="anchored" data-anchor-id="p-value-using-randomization">P-Value using Randomization</h2>
<p>We can also use our simulation skills to determine the null distribution!</p>
<p>. . .</p>
<p>We can randomly rearrange (or <strong>permute</strong>) data so that no relationship is present, e.g.&nbsp;assigning group membership to the participants randomly. In this case, <span class="math inline">\(H_0\)</span> should thus be true.</p>
<p>We would do this a large amount of times (e.g.&nbsp;10000), calculate the <em>t</em> statistics for each iteration, and draw a histogram to show the distribution.</p>
</section>
<section id="p-values-using-randomization-2" class="level2">
<h2 class="anchored" data-anchor-id="p-values-using-randomization-2">P-Values using Randomization 2</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># create function to shuffle BMI data</span></span>
<span id="cb13-2"><a href="#cb13-2"></a>shuffleBMIstat <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb13-3"><a href="#cb13-3"></a>  bmiDataShuffled <span class="ot">&lt;-</span> </span>
<span id="cb13-4"><a href="#cb13-4"></a>    NHANES_sample <span class="sc">%&gt;%</span></span>
<span id="cb13-5"><a href="#cb13-5"></a>    <span class="fu">select</span>(BMI, PhysActive) <span class="sc">%&gt;%</span></span>
<span id="cb13-6"><a href="#cb13-6"></a>    <span class="fu">mutate</span>(</span>
<span id="cb13-7"><a href="#cb13-7"></a>      <span class="at">BMI =</span> <span class="fu">sample</span>(BMI) <span class="co">#randomly shuffle BMI values</span></span>
<span id="cb13-8"><a href="#cb13-8"></a>    )</span>
<span id="cb13-9"><a href="#cb13-9"></a>  <span class="co"># compute the difference</span></span>
<span id="cb13-10"><a href="#cb13-10"></a>  simResult <span class="ot">&lt;-</span> <span class="fu">t.test</span>( <span class="co">#t.test function is more convenient than pt function!</span></span>
<span id="cb13-11"><a href="#cb13-11"></a>    BMI <span class="sc">~</span> PhysActive,</span>
<span id="cb13-12"><a href="#cb13-12"></a>    <span class="at">data =</span> bmiDataShuffled,</span>
<span id="cb13-13"><a href="#cb13-13"></a>  )</span>
<span id="cb13-14"><a href="#cb13-14"></a>  <span class="fu">return</span>(simResult<span class="sc">$</span>statistic)</span>
<span id="cb13-15"><a href="#cb13-15"></a>}</span>
<span id="cb13-16"><a href="#cb13-16"></a><span class="co"># run function 5000 times and save output</span></span>
<span id="cb13-17"><a href="#cb13-17"></a>nRuns <span class="ot">&lt;-</span> <span class="dv">5000</span></span>
<span id="cb13-18"><a href="#cb13-18"></a>meanDiffSimDf <span class="ot">&lt;-</span> <span class="fu">tibble</span>(<span class="at">meanDiffSim =</span> <span class="fu">replicate</span>(nRuns, <span class="fu">shuffleBMIstat</span>()))</span>
<span id="cb13-19"><a href="#cb13-19"></a><span class="co">#run t test of actual data</span></span>
<span id="cb13-20"><a href="#cb13-20"></a>bmtTTest <span class="ot">&lt;-</span> </span>
<span id="cb13-21"><a href="#cb13-21"></a>  <span class="fu">t.test</span>(</span>
<span id="cb13-22"><a href="#cb13-22"></a>  BMI <span class="sc">~</span> PhysActive,</span>
<span id="cb13-23"><a href="#cb13-23"></a>  <span class="at">data =</span> NHANES_sample,</span>
<span id="cb13-24"><a href="#cb13-24"></a>  <span class="at">alternative =</span> <span class="st">"greater"</span></span>
<span id="cb13-25"><a href="#cb13-25"></a>)</span>
<span id="cb13-26"><a href="#cb13-26"></a><span class="co">#compare actual data with simulation</span></span>
<span id="cb13-27"><a href="#cb13-27"></a>bmiPvalRand <span class="ot">&lt;-</span> </span>
<span id="cb13-28"><a href="#cb13-28"></a>  <span class="fu">mean</span>(meanDiffSimDf<span class="sc">$</span>meanDiffSim <span class="sc">&gt;=</span> bmtTTest<span class="sc">$</span>statistic)</span>
<span id="cb13-29"><a href="#cb13-29"></a></span>
<span id="cb13-30"><a href="#cb13-30"></a><span class="co">#plot everything</span></span>
<span id="cb13-31"><a href="#cb13-31"></a>plot <span class="ot">=</span> meanDiffSimDf <span class="sc">%&gt;%</span> </span>
<span id="cb13-32"><a href="#cb13-32"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(meanDiffSim)) <span class="sc">+</span></span>
<span id="cb13-33"><a href="#cb13-33"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span> <span class="dv">200</span>) <span class="sc">+</span></span>
<span id="cb13-34"><a href="#cb13-34"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> bmtTTest<span class="sc">$</span>statistic, <span class="at">color =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb13-35"><a href="#cb13-35"></a>  <span class="fu">xlab</span>(<span class="st">"T stat: BMI difference between groups"</span>) <span class="sc">+</span></span>
<span id="cb13-36"><a href="#cb13-36"></a>  <span class="fu">geom_histogram</span>(</span>
<span id="cb13-37"><a href="#cb13-37"></a>    <span class="at">data =</span> meanDiffSimDf <span class="sc">%&gt;%</span> </span>
<span id="cb13-38"><a href="#cb13-38"></a>      <span class="fu">filter</span>(meanDiffSim <span class="sc">&gt;=</span> bmtTTest<span class="sc">$</span>statistic), </span>
<span id="cb13-39"><a href="#cb13-39"></a>    <span class="fu">aes</span>(meanDiffSim), </span>
<span id="cb13-40"><a href="#cb13-40"></a>    <span class="at">bins =</span> <span class="dv">200</span>, </span>
<span id="cb13-41"><a href="#cb13-41"></a>    <span class="at">fill =</span> <span class="st">"gray"</span></span>
<span id="cb13-42"><a href="#cb13-42"></a>  )</span>
<span id="cb13-43"><a href="#cb13-43"></a><span class="fu">print</span>(plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="p-values-using-randomization-3" class="level2">
<h2 class="anchored" data-anchor-id="p-values-using-randomization-3">P-Values using Randomization 3</h2>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="W7_Hypothesis_files/figure-html/unnamed-chunk-9-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The blue line is the observed <em>t</em> statistic. We can calculate a <em>p</em>-Value by counting how many of the simulated <em>t</em>-values are at least as extreme as our observed one and dividing it by the number of simulations. The p-value obtained from randomization (0.000000) is kind of similar to the one obtained using the t distribution (0.000075).</p>
<p>. . .</p>
<p>Using simulations to get the null distribution can be helpful if the assumptions (normal distribution in each group) are violated or if we don’t know the theoretical distribution of the test statistic!</p>
<div class="notes">
<p>exchangeability: We can use permutations if all observsations are distributed in the same way, such that we can shuffle them without changing the overall distribution.</p>
<p>Not the case if we have dependent observations, e.g.&nbsp;siblings…</p>
</div>
</section>
<section id="step-6-assess-the-statistical-significance-of-the-result" class="level2">
<h2 class="anchored" data-anchor-id="step-6-assess-the-statistical-significance-of-the-result">Step 6: Assess the “Statistical Significance” of the Result</h2>
<p>Is the <em>p</em>-value determined small enough to reject the null hypothesis (and thus conclude that the alternative hypothesis is true)?</p>
<p>. . .</p>
<p>Traditionally, we reject <span class="math inline">\(H_0\)</span> if the <em>p</em>-value is less than <strong>0.05</strong>. (Fisher’s approach)</p>
<p>. . .</p>
<p>(Either there is an effect/<span class="math inline">\(H_A\)</span> is true or there is a small chance (5%) that there is actually no effect but we coincidentally found such a large value –&gt; false positive)</p>
<p>. . .</p>
<p>Neyman-Pearson approach: In the long run, we will know how often we are wrong:</p>
<ul>
<li><p><span class="math inline">\(\alpha = .05\)</span> (false positives or Type I error: We reject <span class="math inline">\(H_0\)</span> although it is correct),</p></li>
<li><p><span class="math inline">\(\beta = .2\)</span> (false negatives or Type II error: We accept <span class="math inline">\(H_0\)</span> although it is wrong),</p></li>
<li><p>We will be correct if we reject <span class="math inline">\(H_0\)</span> when it is wrong (there is actually a difference/an effect) or if we do not reject <span class="math inline">\(H_0\)</span> when it is correct (and there is no difference between groups).</p></li>
</ul>
<p>In both cases, a significance level of <span class="math inline">\(\alpha = .05\)</span> is usually used.</p>
<div class="notes">
<p>How much evidence do we require?</p>
<p>0.05 Fisher never intended it to be fixed</p>
<p>Before computers, tables were used, and all tables had .05 in it!</p>
<p>Fisher: Evidence for hypothesis, NP: long-run error rate</p>
</div>
</section>
<section id="what-does-a-significant-result-mean" class="level2">
<h2 class="anchored" data-anchor-id="what-does-a-significant-result-mean">What does a significant result mean?</h2>
<p>There is a lot of discussion about the usefulness of using <span class="math inline">\(\alpha = .05\)</span> as well as about the interpretation of a significant result/certain <em>p</em>-value!</p>
<p>. . .</p>
<div class="incremental">
<p>A <em>p</em>-value of .01 does….</p>
<ul class="incremental">
<li><p>NOT mean that the probability that <span class="math inline">\(H_0\)</span> is true is 1%!</p>
<ul class="incremental">
<li>We tested <span class="math inline">\(P(data|H_0)\)</span> not <span class="math inline">\(P(H_0|data)\)</span>!</li>
</ul></li>
<li><p>NOT mean that the probability that you’re making a wrong decision is 1%!</p>
<ul class="incremental">
<li>This would also be <span class="math inline">\(P(H_0|data)\)</span>! p-values are probabilities of data (under <span class="math inline">\(H_0\)</span>), not probabilities of hypotheses! And we cannot easily use Bayes to turn the condition because we would need additional information like the prior probability of an alternative hypothesis being true.</li>
</ul></li>
<li><p>NOT mean that you would get the same significance 99% of the time if you repeated the study.</p>
<ul class="incremental">
<li>The <em>p</em>-value is a statement about the likelihood of <em>one particular dataset</em> under the null.</li>
</ul></li>
<li><p>NOT mean that you found a practically important effect.</p>
<ul class="incremental">
<li>Difference between <em>statistical significance</em> and <em>practical significance</em>! Effect sizes more important. (Statistical significance depends on sample size!)</li>
</ul></li>
</ul>
</div>
</section>
<section id="multiple-testing" class="level2">
<h2 class="anchored" data-anchor-id="multiple-testing">Multiple Testing</h2>
<p>Nowadays, we often have huge datasets in neuroscience, e.g.&nbsp;collecting brain imaging data of thousands of voxels or quantifying the entire genome.</p>
<p>. . .</p>
<p>Let’s look at <em>genome-wide associations studies</em> (GWAS). We have more than a million places in where the genome could differ. If we want to know whether <em>schizophrenia</em> was associated with <em>any</em> of these differences, we would do ~1.000.000 tests! If we simply used <span class="math inline">\(\alpha \le .05\)</span> as a threshold, we would get a lot of (<span class="math inline">\(1000000 * .05 = 500\)</span>!) false positives, even if no true effect is present at all.</p>
<p>. . .</p>
<p>In this case, we have a lot of dependent tests, which form a <em>family of tests</em>. In such a case, we need to control the <em>family-wise error rate</em>, e.g.&nbsp;by fixing it to a total of <span class="math inline">\(\alpha \le .05\)</span> (i.e.&nbsp;the probability of making <em>any</em> Type I error in our study is controlled at .05).</p>
<p>. . .</p>
<p>One option is to use the <strong>Bonferroni correction</strong>, in which we divide .05 by the number of tests (e.g.&nbsp;1.000.000) and use the new value (<span class="math inline">\(\alpha \le .000005\)</span>) as threshold for each individual test.</p>
<p>. . .</p>
<p>This is extremely conservative and often results in false negative test results.</p>
<p>. . .</p>
<p>For an interesting example what can happen when not correcting for multiple comparisons, see this dead fish showing significant brain activity (<a href="https://www.psychology.mcmaster.ca/bennett/psy710/readings/BennettDeadSalmon.pdf">Link</a>).</p>
</section>
</section>
<section id="quantifying-effects-and-designing-studies" class="level1 page-columns page-full">
<h1>Quantifying Effects and Designing Studies</h1>
<p>So far, we have discussed how to use data to test hypotheses, which results in either rejecting or failing to reject the null hypothesis <span class="math inline">\(H_0\)</span>.</p>
<p>However, looking only at the significance (reject or not) ignores the <em>uncertainty</em> we have about the conclusion. We can calculate <strong>confidence intervals</strong> to quantify uncertainty about our estimates.</p>
<p>Furthermore, we would also like to know how large (i.e., practically relevant) an effect is. For this aim, we will calculate <strong>effect sizes</strong>.</p>
<p>Finally, we will also talk about <strong>statistical power</strong>, which tells us how likely we are to find a true effect given our experimental setup (i.e., mostly sample size).</p>
<section id="confidence-intervals" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="confidence-intervals">Confidence Intervals</h2>
<p>Single value statistic (e.g.&nbsp;<em>t</em>-value, mean…) = point estimate</p>
<p>. . .</p>
<p>We know from the sampling error discussion that each point estimate comes with some uncertainty, described by the standard error.</p>
<p>Remember, the SEM (standard error of the mean) was calculated with the sample standard deviation <span class="math inline">\(\hat{\sigma}\)</span> and the square root of the sample size <span class="math inline">\(n\)</span>:</p>
<p><span class="math display">\[SEM = \frac{\hat{\sigma}}{\sqrt{n}}\]</span></p>
<p><span class="math inline">\(n\)</span> is generally under our control (<span class="math inline">\(\hat{\sigma}\)</span> is unknown but fixed*), and we can thus <em>decrease</em> our uncertainty by <em>increasing</em> the sample size.</p>

<div class="no-row-height column-margin column-container"><div class="margin-aside">
<p>* In practice, we <em>can</em> acquire homogeneous samples (like students) or heterogeneous samples (like extreme groups). This will, however, influence our generalizability.</p>
</div></div><p>. . .</p>
<p>We can more directly describe our uncertainty with <strong>confidence intervals (CI)</strong>, which provides a range of values for our parameter estimate that are consistent with our data! The wider the CI, the more uncertain we are about our estimate.</p>
</section>
<section id="confidence-intervals-2" class="level2">
<h2 class="anchored" data-anchor-id="confidence-intervals-2">Confidence Intervals 2</h2>
<p>Because the CI depends on the SEM, which decreases with sample size, the CI also gets narrower with increasing sample size:</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="W7_Hypothesis_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="notes">
<p>if we sample the whole population, we <em>know</em> the population parameter and there is no uncertainty at all!</p>
<p>Range of possible values (estimate) in concordance with the data</p>
</div>
</section>
<section id="confidence-intervals-3" class="level2">
<h2 class="anchored" data-anchor-id="confidence-intervals-3">Confidence Intervals 3</h2>
<p>Just like <em>p</em>-values, confidence intervals can be confusing because they are counter-intuitive: A 95% CI for a statistic does NOT mean that we can have 95% confidence that the true parameter falls within this interval!</p>
<p>It is, again, the long-run probability: It will contain the true population parameter 95% of the time in the long-run.</p>
<p>Let’s sample 100 times with <span class="math inline">\(n = 250\)</span> from the NHANES data and calculate CIs. We use the NHANES mean as true score (dashed line) and check how often the CI misses to include it: 5% of the time.</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="W7_Hypothesis_files/figure-html/unnamed-chunk-12-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="notes">
<p>but: new experiment = new estimate (under exact same conditions would work)!</p>
</div>
</section>
<section id="calculating-the-ci" class="level2">
<h2 class="anchored" data-anchor-id="calculating-the-ci">Calculating the CI</h2>
<p>We calculate the CI as follows:</p>
<p><span class="math inline">\(CI = \text{point estimate} \pm \text{critical value} * \text{standard error}\)</span></p>
<p>The “critical value” depends on the sampling distribution. Most of the time, we will use the normal distribution.</p>
<div class="notes">
<p>CI depends on the confidence level (95%, critical value), the sample size and the variability in the data (both SE)</p>
</div>
</section>
<section id="ci-using-the-normal-distribution" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="ci-using-the-normal-distribution">CI using the Normal Distribution</h2>
<p>The critical value are the values of the standard normal distribution that capture 95% (in case of a 95% CI) of the distribution, i.e.&nbsp;the 2.5<sup>th</sup> and 97.5<sup>th</sup> percentile.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qnorm</span>(<span class="at">p=</span><span class="fu">c</span>(.<span class="dv">025</span>,.<span class="dv">975</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -1.959964  1.959964</code></pre>
</div>
</div>
<p>The CI of the mean would thus be:</p>
<p><span class="math inline">\(CI = \bar{X} \pm 1.96*SE\)</span></p>
<p>Our mean weight in the NHANES sample was 79.92 kg and the SE was <span class="math inline">\(\frac{SD_{weight}}{\sqrt{n}} = 1.35\)</span> (for a random <span class="math inline">\(n=250\)</span> subsample)*.</p>

<div class="no-row-height column-margin column-container"><div class="margin-aside">
<p>* In practice, both the mean and standard deviation will be subject to sampling error, i.e., CIs will differ both in location and size despite having the same sample size (cf.&nbsp;plot 2 slides ago)</p>
</div></div><p>The lower boundary of the CI of the mean would then be <span class="math inline">\(CI = 79.92 - 1.96 * 1.35 = 77.28\)</span> and the upper <span class="math inline">\(CI = 79.92 + 1.96 * 1.35 = 82.56\)</span>. We would write this as [77.28, 82.56].</p>
</section>
<section id="ci-using-the-t-distribution" class="level2">
<h2 class="anchored" data-anchor-id="ci-using-the-t-distribution">CI using the <em>t</em> Distribution</h2>
<p>If we don’t know the population standard deviation, which is usually the case, it is more appropriate to use the <em>t</em> distribution.</p>
<p>In this case, we use the critical value of the <em>t</em> distribution:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">qt</span>(<span class="at">p =</span> <span class="fu">c</span>(.<span class="dv">025</span>, .<span class="dv">975</span>), <span class="dv">250</span>) <span class="co">#t distribution depends on sample size! =&gt; n = 250</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -1.969498  1.969498</code></pre>
</div>
</div>
<p>For the NHANES weight example, the CI would be: <span class="math inline">\(79.92 \pm 1.97 * 1.35 = [77.15, 82.58]\)</span>.</p>
<p>The CI for a <em>t</em> distribution is always larger than for a normal distribution. In practice, this difference is negligible at <span class="math inline">\(n \ge 30\)</span> (1.984 vs.&nbsp;1.960).</p>
<div class="notes">
<p>The <em>t</em> distribution is wider than the normal distribution (especially for smaller samples), which means that the CI will be slightly wider -&gt; extra uncertainty smaller samples.</p>
<p>population parameter hast a fixed value, so it either falls into CI or not. (Doesn’t make sense to talk about probability of it)</p>
</div>
</section>
<section id="ci-using-the-bootstrap" class="level2">
<h2 class="anchored" data-anchor-id="ci-using-the-bootstrap">CI using the Bootstrap</h2>
<p>If we can’t assume normality or don’t know the sampling distribution, we can also use the <strong>bootstrap</strong> to compute the CI.</p>
<p>. . .</p>
<p>Reminder: bootstrap = resampling with replacement, using this distribution as the sampling distribution!</p>
<p>. . .</p>
<p>If we use an R function for bootstrapping (<code>boot()</code>), we get CI estimates that are fairly close to the ones calculated:</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code>BOOTSTRAP CONFIDENCE INTERVAL CALCULATIONS
Based on 1000 bootstrap replicates

CALL : 
boot.ci(boot.out = bs, type = "perc")

Intervals : 
Level     Percentile     
95%   (78.41, 83.81 )  
Calculations and Intervals on Original Scale</code></pre>
</div>
</div>
<p>. . .</p>
<p><br>
</p>
<table class="caption-top table">
<thead>
<tr class="header">
<th>Method</th>
<th>CI</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>normal distribution</td>
<td>[77.28, 82.56]</td>
</tr>
<tr class="even">
<td><em>t</em> distribution</td>
<td>[77.15, 82.58]</td>
</tr>
<tr class="odd">
<td>bootstrap</td>
<td>[78.41, 83.81]</td>
</tr>
</tbody>
</table>
<p>. . .</p>
<p><strong>Note</strong>: CIs from the normal and <em>t</em> distribution are always symmetrical around the mean (here: 79.92), bootstrapped intervals need not be.</p>
</section>
<section id="relationship-of-cis-to-hypothesis-tests" class="level2">
<h2 class="anchored" data-anchor-id="relationship-of-cis-to-hypothesis-tests">Relationship of CIs to Hypothesis Tests</h2>
<p>If the CI <em>does not</em> include the value of the null hypothesis (usually 0), then the associated two-sided one sample test would be significant.</p>
<p>. . .</p>
<p>If we want to compare two conditions, it gets trickier.</p>
<ul>
<li><p>If each mean is contained within the CI of the other mean, then there’s <em>definitely no</em> significant difference.</p></li>
<li><p>If there is no overlap between CIs, then there is <em>certainly</em> a significant difference (two-sidedly).</p></li>
<li><p>If the CIs overlap (but don’t contain the other mean), it depends on the relative variability of the two variables</p></li>
</ul>
<p>In general, avoid this “eyeball test” and look at the <em>p</em> value!</p>
<p><strong>Note</strong>: Slightly overlapping CIs can still be significant (e.g., one sided hypothesis)</p>
</section>
<section id="effect-sizes" class="level2">
<h2 class="anchored" data-anchor-id="effect-sizes">Effect Sizes</h2>
<p>Practical significance!</p>
<p>We need a standard way to describe the size of an effect.</p>
<p>An <strong>effect size</strong> is a standardized measurement that compares the size of an effect to e.g.&nbsp;the variability of the statistic. This is also referred to as <em>signal-to-noise ratio.</em></p>
<p>There are many different variants of effect sizes!</p>
</section>
<section id="cohens-d" class="level2">
<h2 class="anchored" data-anchor-id="cohens-d">Cohen’s <em>d</em></h2>
<p>Cohen’s <em>d</em> is used to quantify the difference between <em>two</em> means, in terms of their SD:</p>
<p><span class="math display">\[d = \frac{\bar{X_1} - \bar{X}_2}{s}\]</span></p>
<p>where <span class="math inline">\(\hat{X_1}\)</span> and <span class="math inline">\(\hat{X_2}\)</span> are the means of the two groups and <span class="math inline">\(s\)</span> is the pooled SD:</p>
<p><span class="math display">\[s = \sqrt{\frac{(n_1 - 1)s^2_1 + (n_2 - 1)s^2_2 }{n_1 +n_2 -2}}\]</span></p>
<p>which is a combination of both groups’ SDs (<span class="math inline">\(s_1\)</span> and <span class="math inline">\(s_2\)</span>) weighted by their sample size (<span class="math inline">\(n_1\)</span> and <span class="math inline">\(n_2\)</span>).</p>
<div class="notes">
<p>Note that this is very similar to the <em>t</em> statistic, only the denominator differs:<br>
<em>t</em> = SEM, <em>d</em> = SD of the data<br>
–&gt; d will not grow with sample size but remain stable!</p>
</div>
</section>
<section id="cohens-d-2" class="level2">
<h2 class="anchored" data-anchor-id="cohens-d-2">Cohen’s <em>d</em> 2</h2>
<p>There is a commonly used interpretation of Cohen’s d (although it is criticized to use these cutoffs!):</p>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>Interpetation of Cohen’s d</caption>
<thead>
<tr class="header">
<th style="text-align: left;">d</th>
<th style="text-align: left;">Interpretation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">0.0 - 0.2</td>
<td style="text-align: left;">negligible</td>
</tr>
<tr class="even">
<td style="text-align: left;">0.2 - 0.5</td>
<td style="text-align: left;">small</td>
</tr>
<tr class="odd">
<td style="text-align: left;">0.5 - 0.8</td>
<td style="text-align: left;">medium</td>
</tr>
<tr class="even">
<td style="text-align: left;">&gt; 0.8</td>
<td style="text-align: left;">large</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>. . .</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>effect <span class="ot">=</span> NHANES_sample <span class="sc">%&gt;%</span> effsize<span class="sc">::</span><span class="fu">cohen.d</span>(Height <span class="sc">~</span> Gender, <span class="at">data=</span>.) <span class="sc">%&gt;%</span> {.<span class="sc">$</span>estimate} <span class="sc">%&gt;%</span> <span class="fu">abs</span>() <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>. . .</p>
<p>Even with a large effect of <span class="math inline">\(d = 1.78\)</span> (as in the NHANES data set), the distributions still overlap greatly!</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="W7_Hypothesis_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="notes">
<p>Small effect but huge total impact: Psychosocial stress due to Covid-19 (deployed to a whole population)</p>
</div>
</section>
<section id="pearsons-r" class="level2">
<h2 class="anchored" data-anchor-id="pearsons-r">Pearson’s <em>r</em></h2>
<p>Pearson’s <em>r</em> is a <em>correlation coefficient</em>, and thus a measure of the strength of a <em>linear</em> relationship between <em>two continuous</em> variables.</p>
<p><em>r</em> can vary from -1 to 1: -1 is a perfect negative relationship, 0 no (linear) relationship, and 1 a perfect positive relationship. Try your skills eye-balling the size of a correlation: <a href="https://www.guessthecorrelation.com/" class="uri">https://www.guessthecorrelation.com/</a></p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="W7_Hypothesis_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="notes">
<p>more on correlations later in the semester!</p>
</div>
</section>
<section id="odds-ratio" class="level2">
<h2 class="anchored" data-anchor-id="odds-ratio">Odds Ratio</h2>
<p>For binary variables, the odds ratio is a useful effect size.</p>
<p><strong>Odds</strong> describes the relative likelihood of some event happening versus not happening:</p>
<p><span class="math display">\[
\text{odds of A} = \frac{P(A)}{P(\neg{A})}
\]</span></p>
<p><strong>Odds ratio</strong> is simply the ratio of two odds, i.e.&nbsp;<span class="math inline">\(\frac{\text{odds of A}}{\text{odds of B}}\)</span>.</p>
<p>. . .</p>
<p>Example:</p>
<div class="cell">
<div class="cell-output-display">
<table class="caption-top table table-sm table-striped small">
<caption>Lung cancer occurrence separately for current smokers and those who have never smoked</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Status</th>
<th style="text-align: right;">NeverSmoked</th>
<th style="text-align: right;">CurrentSmoker</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">No Cancer</td>
<td style="text-align: right;">2883</td>
<td style="text-align: right;">3829</td>
</tr>
<tr class="even">
<td style="text-align: left;">Cancer</td>
<td style="text-align: right;">220</td>
<td style="text-align: right;">6784</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math display">\[OR = \frac{\frac{220}{220+2883}}{\frac{6784}{6784+3829}} = 23.22 \]</span></p>
<p>The odds ratio of 23.22 tells us that the odds of lung cancer in smokers are roughly 23x higher than that of non-smokers!</p>
</section>
<section id="statistical-power" class="level2">
<h2 class="anchored" data-anchor-id="statistical-power">Statistical Power</h2>
<p>Remember: <em>Type I</em> and <em>Type II error</em>!</p>
<p>Tolerance for Type I errors set to <span class="math inline">\(\alpha = 0.05\)</span>, which is very low –&gt; we want to avoid this error!</p>
<p>What about Type II errors?</p>
<p>. . .</p>
<p>Type II = failing to reject <span class="math inline">\(H_0\)</span> although an effect exists (often set at <span class="math inline">\(\beta = 0.20\)</span>, i.e., a statistical power of <span class="math inline">\(80\%\)</span>).</p>
<p>But <span class="math inline">\(\beta\)</span> also depends on the effect size: The likelihood of finding a large effect is higher than finding a small effect (at constant <span class="math inline">\(n\)</span>)!</p>
<p>. . .</p>
<p><strong>Statistical power</strong> is the complement of the Type II error:<br>
The likelihood of finding a positive result given that it exits!<br>
<span class="math display">\[power = 1 - \beta\]</span></p>
<p>. . .</p>
<p>Statistical power is affected by three factors:</p>
<ul>
<li><p>sample size (larger <em>n</em> = more power)</p></li>
<li><p>effect size (larger effect = more power)</p></li>
<li><p>Type I error rate (smaller Type I error = less power)</p></li>
</ul>
<div class="notes">
<p>Type I: false positives</p>
<p>Type II: false negatives</p>
</div>
</section>
<section id="statistical-power-2" class="level2">
<h2 class="anchored" data-anchor-id="statistical-power-2">Statistical Power 2</h2>
<p>Here, we can see how these three factors influence the power (i.e.&nbsp;the proportion of significant results found):</p>
<div class="cell">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="W7_Hypothesis_files/figure-html/unnamed-chunk-22-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>The black dotted line denotes the standard 80% power that is often aimed at.</p>
<p>. . .</p>
<p>Even with <span class="math inline">\(n = 96\)</span>, we have only little power to detect a small effect (<span class="math inline">\(d = 0.2\)</span>): Only ~25% of studies would find the true effect. This means doing this study would be futile, we would likely fail to find the true effect.</p>
<p>. . .</p>
<p>Therefore, we would do a <strong>power analysis</strong> before we even run the study - to determine the necessary sample size for a well-powered study that would be able to find an effect if the effect is true.</p>
<p>Furthermore, positive findings from an underpowered study are more likely to be false positive!</p>
<div class="notes">
<p>More on that in Chapter 18 of ST21!</p>
</div>
</section>
</section>
<section id="thanks" class="level1">
<h1>Thanks!</h1>
<p>That’s it for today!</p>
<p><strong>Until next week, you should have decided for a data set for your analysis!</strong></p>
<p>Next week, we will start with implementing analyses in R.</p>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>